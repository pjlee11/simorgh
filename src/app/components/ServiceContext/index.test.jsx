import React from 'react';
import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';
import { ServiceContextProvider, ServiceContextConsumer } from './index';

describe('ServiceContext', () => {
  const testBrandNameWithServiceContext = service => {
    shouldMatchSnapshot(
      `should have a brand name for ${service}`,
      <ServiceContextProvider service={service}>
        <ServiceContextConsumer>
          {props => <span>{props.brandName}</span>}
        </ServiceContextConsumer>
      </ServiceContextProvider>,
    );
  };

  testBrandNameWithServiceContext('news');
  testBrandNameWithServiceContext('persian');
  testBrandNameWithServiceContext('default');
});
