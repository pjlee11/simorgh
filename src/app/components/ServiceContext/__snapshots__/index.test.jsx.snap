// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceContext should have a brand name for default 1`] = `
<span>
  Default Brand Name
</span>
`;

exports[`ServiceContext should have a brand name for news 1`] = `
<span>
  BBC News
</span>
`;

exports[`ServiceContext should have a brand name for persian 1`] = `
<span>
  BBC News فارسی
</span>
`;
